src/lib/utils/__tests__/index-test.js

```diff
@@ -149,6 +149,82 @@ describe('utils.renderTemplate', () => {
     expect(actual).toBe(expectation);
   });
 
+  it('expect to process templates as string using a partial', () => {
+    const templateKey = 'test';
+    const templates = {
+      test: '{{> partial}}',
```

Comment:
worth nothing this can accidentally cause infinite loops if `test: {{> test}}` or partial calls test itself. 
Is there a way it can be prevented?

Reply:
As far as I can see this is a general issue with Mustache partials more than it is a problem with this PR, and would more properly be handled in the Hogan code, if anywhere.

Still, it might be possible to change the InstantSearch.js renderTemplate() function to keep track of what templates it has rendered and prevent any template from being used more than a few times in a given call chain, or alternatively to keep track of a total recursion depth and impose some limit, if you're interested in me attempting it at the expense of the added complexity.